kgforge/specializations/stores/bluebrain_nexus.py

```diff
@@ -124,80 +123,57 @@ def register_callback(task: Task):
             resources,
-            function_name=self._register_many.__name__,
+            function_name=fc_name,
             exception=RegistrationError,
             id_required=False,
             required_synchronized=False,
             execute_actions=True,
         )
-        params_register = copy.deepcopy(self.service.params.get("register", {}))
 
-        self.service.batch_request(
+        BatchRequestHandler.batch_request_on_resources(
+            service=self.service,
             resources=verified,
-            action=BatchAction.CREATE,
             callback=register_callback,
-            error_type=RegistrationError,
+            prepare_function=prepare_methods.prepare_create,
             schema_id=schema_id,
-            params=params_register,
         )
 
     def _register_one(self, resource: Resource, schema_id: str) -> None:
-        context = self.model_context() or self.context
-        data = as_jsonld(
-            resource,
-            "compacted",
-            False,
-            model_context=context,
-            metadata_context=None,
-            context_resolver=self.service.resolve_context
-        )
 
-        params_register = copy.deepcopy(self.service.params.get("register", None))
-        identifier = resource.get_identifier()
-
-        if identifier:
-
-            url = Service.add_schema_and_id_to_endpoint(
-                self.service.url_resources, schema_id=schema_id, resource_id=identifier
-            )
+        method, url, resource, exception_, headers, params, payload = prepare_methods.prepare_create(
+            service=self.service, resource=resource
+        )
 
-            response = requests.put(
-                url,
-                headers=self.service.headers,
-                data=json.dumps(data, ensure_ascii=True),
-                params=params_register,
-                timeout=REQUEST_TIMEOUT
-            )
-        else:
+        response = requests.request(
+            method=method,
+            url=url,
+            headers=headers,
+            data=json.dumps(payload, ensure_ascii=True),
+            params=params,
+            timeout=REQUEST_TIMEOUT
+        )
 
-            url = Service.add_schema_and_id_to_endpoint(
-                self.service.url_resources, schema_id=schema_id, resource_id=None
-            )
+        catch_http_error_nexus(response, exception_)
 
-            response = requests.post(
-                url,
-                headers=self.service.headers,
-                data=json.dumps(data, ensure_ascii=True),
-                params=params_register,
-                timeout=REQUEST_TIMEOUT
-            )
-        catch_http_error_nexus(response, RegistrationError)
+        data = payload
 
         response_json = response.json()
         resource.id = response_json["@id"]
         # If resource had no context, update it with the one provided by the store.
         if not hasattr(resource, "context"):
```

Comment:
about this id/context addition to the resource when registering... it doesn't seem like a extreme change to make one general callback, I would say ... go for it if you already thought of a way

Reply:
What do you mean by a general callback?

Reply:
there is a default callback method, that is shared within all the possible resource tasks, that is what I meant, why not to move the option of adding id/context there, to have one single method?

Reply:
The setting of the resource.id and context look harmless at first glance so I could see them happening for any resource operation. But are they really harmless?

First there's the question of harmonising the register_one and register_many callbacks before even thinking of making them the same as the other operations'